VisualStudio/src/CookingPatches.cs

```diff
@@ -9,39 +9,27 @@ internal class CookingPotItem_DoSpecialActionFromInspectMode
     {
         internal static bool Prefix(CookingPotItem __instance)
         {
-            //Implementation.Log("CookingPotItem -- DoSpecialActionFromInspectMode");
-            if (__instance.GetCookingState() == CookingPotItem.CookingState.Cooking) //patch does not apply while cooking
+            float waterAmount = __instance.m_LitersWaterBeingBoiled;
+            if (__instance.GetCookingState() == CookingPotItem.CookingState.Ready && waterAmount > 0) //patch does not apply while cooking
             {
-                return true;
-            }
-
-            float waterAmount = WaterUtils.GetWaterAmount(__instance);
-            if (waterAmount <= 0) //There must be some water for this to apply
-            {
-                return true;
-            }
-
-            //sets to true if the item has been boiled
-            bool potable = __instance.GetCookingState() == CookingPotItem.CookingState.Ready;
-
-            GearItem gearItem = __instance.GetComponent<GearItem>();
-
-            WaterSupply waterSupply = gearItem.m_WaterSupply; //Gets the cooking pot's water supply component
-            if (waterSupply == null)//if it doesn't exist
-            {
-                waterSupply = gearItem.gameObject.AddComponent<WaterSupply>();//create one
-                gearItem.m_WaterSupply = waterSupply;//and assign it
+                GearItem gearItem = __instance.GetComponent<GearItem>();
+                WaterSupply waterSupply = gearItem.m_WaterSupply;
+                if (waterSupply == null)//if it doesn't exist
+                {
+                    waterSupply = gearItem.gameObject.AddComponent<WaterSupply>();//create one
+                    gearItem.m_WaterSupply = waterSupply;//and assign it
+                }
+                float waterVolumeToDrink = GameManager.GetPlayerManagerComponent().CalculateWaterVolumeToDrink(waterAmount);
+                __instance.m_LitersWaterBeingBoiled -= waterVolumeToDrink;
+                waterSupply.m_VolumeInLiters = waterAmount;
+                waterSupply.m_WaterQuality = LiquidQuality.Potable;
```

Comment:
This was the small bug with Potable vs nonPotable. Will be corrected in the next commit